core/loadpoint.go

```diff
@@ -1500,6 +1541,13 @@ func (lp *LoadPoint) Update(sitePower float64, cheap bool, batteryBuffered bool)
 		err = lp.setLimit(targetCurrent, required)
 	}
 
+	// WakeUp checks
+	if lp.enabled && lp.status == api.StatusB &&
```

Comment:
Ich schau grad alles nochmal durch. Ich glaub hier haben wir einen echten Klemmer. Mit der Logik können wir aktuell nur Autos aufwecken, die erkannt werden und einen SoC liefern. Das kann beabsichtigt sein. Müssten wir an der Stelle aber nicht eigentlich _jedes_ Auto aufwecken dass nicht laden will, zumindest über den Charger?

Reply:
wenn das Auto keinen SoC liefert dann ist er kleiner 100.
nur wenn das Auto nicht erkannt wird wird es nicht geweckt. 
`lp.vehicle != nil`
sollten wir es rausnehmen. Aber darf es vorkommen das ein unbekanntes Auto lädt?

Reply:
Ja, das das mss gehen (=Gastfahrzeuge). Nicht laden ist über `mode=OFF` abgedeckt. Das hat allerdings ein Folgeproblem: unbekannte Fahrzeuge werden- wenn der Akku voll ist- auch über den Charger erneut aufgeweckt. Das scheint mir keine gute Idee zu sein?

/cc @premultiply

Reply:
Würde aber auch nur ein mal pro Aktivierung passieren. `enabled=true`
Oder wir müssten wieder ein Flag speichern damit es nur ein mal pro Verbundenem Auto passiert und diesen zurücksetzen wenn das Auto lädt oder es getrennt wird, was aber manchmal beim wakeup über den charger passiert.

Können wir dann ein Problem bekommen? Wenn ein auto schläft - es aufgeweckt wird und dann aber der Strom wieder abgeschaltet bevor es lädt. Dann würde es nichtmehr aufgeweckt, da wir es einmal versucht hatten.

Reply:
Darf ich nochmal auf die Phoenix-Implementierung verweisen?
Da steht doch eigentlich alles drin wie es funktioniert.

Reply:
![grafik](https://user-images.githubusercontent.com/4681172/150685776-b65cf28d-28d9-45ac-a412-4c81995dd872.png)

Reply:
Dieser Vorgang wird nach dem Übergang von Status A1 oder B1 in Status B2 maximal einmal ausgeführt.
Der Aktivierungsmodus wird unter diesen Bedingungen wieder ausgeführt.
– Das Fahrzeug wird von der Ladesteuerung getrennt und neu angeschlossen.
– Der Ladevorgang wird durch die Ladestation unterbrochen (z. B. aus Gründen des
Lastmanagements).

Reply:
B1/B2 nutzt uns nichts- denn die Information gibts nicht. Lastmanagement ist genau „enable/disable“. D.h. es wäre richtig, ein unbekanntes Fahrzeug- also auch eines bei dem der Akku voll sein könnte- jeweils erneut über den Charger aufzuwecken.

Konsens?

Reply:
Nein. Doch gibt es.
B1 = StatusB + !enabled
B2 = StatusB + enabled

Entscheidend ist hier der Zustand von enabled. Das d am Ende von enabled ist entscheidend.

Reply:
Also egal ob ein Auto erkannt wurde machen wir einmalig einen reset wenn auf B2 gewechselt wird und dort bleibt.
Aber dein call vom der Auto API testen wir den SOC vorher ob wir ihn wecken wollen oder nicht.
oder?

Reply:
Genau. Wenn "wakedup != true" und nach Status B2 (= StatusB + enabled) gewechselt wird und und nicht spätestens nach Timeout der Zustand C erreicht ist wird das Fahrzeug einmalig geweckt** sowie "wakedup = true" gesetzt. Wird Status A (disconnected) erreicht und/oder ist enabled wieder false wird "wakedup = false" gesetzt.

** Hier würde ich nach der Verfügbarkeit der jeweiligen der APIs mit Priorität auf Charger gehen.
Im Prinzip bei Freigabe des Ladepunkts bis zu zwei parallele Timer starten, der erste schlägt nach 30 Sekunden zu, der andere nach 60 Sekunden. Tritt irgendwann zwischenzeitlich Status C ein werden die Timer abgebrochen.
1. Fall: Charger-Wakeup nach 30 Sekunden, Vehicle-Wakeup nach 60 Sekunden
2. Fall: Charger-Wakeup nach 30 Sekunden
3. Fall: Vehicle-Wakeup nach 30 Sekunden
4. Fall: Keine Wakeup-Methode verfügbar

Reply:
D.h. Im Klartext, dass im pv-Modus bei wechselnder Erzeugung durchaus mehrfacht geweckt wird, auch wenn das Auto voll sein sollte. Wäre das ein Problem?

Reply:
Problem denke ich nicht da es andere Charger so machen. Aber sollten wir es tun??

Reply:
Mehrfach ja, das ist beabsichtigt.
Davor gilt natürlich wenn Ist-SoC < Target-SoC. Bei 100% wird daher nicht geweckt.

Reply:
Ebenso wird nicht geweckt wenn das Fahrzeug selbst die Stromaufnahme beendet hat da dann `enabled` weiterhin `true` bleibt.

Reply:
Ist-SoC < Target-SoC : dann würden wir aber kein enable machen
da wir ja nur beim ändern vom !enabled -> enabled den Timer starten würde er nicht geweckt wenn er aufhört.

Reply:
Der Charger darf während des Wakeups natürlich keinen Disconnect signalisieren. Aber das ist hier nicht relevant.